### DISCUSSION TITLE
Help Needed!

### DISCUSSION THREAD

**Discussion by ncpa0cpl** [react-gjs/renderer#1] [General]
This renderer is still in early development and there are many components that need to be implemented. If you want to help feel free to submit a PR, or ask me any questions!

**Comment by bresnow** [comment]
Hey there... I thought i was the only one crazy enough to tackle typescript wrapped GObjectIntropection in JSX markup. Im currently scaling my Gjsx project so what i have isn't pretty and the examples are half baked but it's legible. And it it's specifically designed for use with the broadwayd server. Which is partly why Im scaling because it's a lot.
Im going to leave a link to the JSX factory / renderer below. 

BTW. I love the way you mapped everything out here. 

https://github.com/bresnow/gjsx-broadway-docker/blob/master/gi_modules/gjsx/runtime.ts

**Reply by bresnow** [reply]
Also also... Im working on a few state "hooks" using Gtk.Builder.new_from_string( JSX_Markup as string ) &&  widgets signalling one another with  gi://Soup websocket connections.

**Comment by kacperpaczos** [comment]
I would like to help, what needs to be done?

**Reply by ncpa0cpl** [reply]
Implementing Elements for the GTK3 Widgets that are not yet done, there are [issues created for some of them](https://github.com/react-gjs/react-gjs-renderer/issues?q=is%3Aissue+is%3Aopen+Implement+GJS+Elements). (there are others not in here that you could try to tackle on as well)

Aside from adding support for these missing widgets, all that really needs to be done on the renderer side is testing and bugfixing. So you could also try playing around with it, and submit issues.

If you have any questions, feel free to ask.

**Comment by rockiger** [comment]
Hello awesome project,

I will play around with it and try to write some documentation. 

For example I had to install `meson`, `gcc` and `gjs-devel` to get it running. I think this is not obvious for web developers - which are the target audience I presume. developers).
